### PR TITLE
Feature/move url retrieval to middleware

### PR DESCRIPTION
**Pull request by neilbmclaughlin**

### CODE REVIEW

**Comment by st3v3nhunt** [comment]
@neilbmclaughlin Can this be rebased and the checks reviewed to make sure they will pass ready for a merge please?

File: test/unit/server.js [hunk 1]
```diff
@@ -15,8 +17,10 @@ describe('Server', () => {
     chai.request(app)
       .get('/gpdetails/12410')
       .end((err, res) => {
-        res.statusCode.should.equal(200);
-        res.text.should.contain('A GP Practice');
+        expect(err).to.be.null;
+        expect(res).to.have.status(200);
+        expect(res).to.be.html;
+        expect(res.text).to.contain('<h2>A GP Practice</h2>');
```
Comment (st3v3nhunt) [inline test/unit/server.js:23]: This feels quite specific to include the html tags

**Comment by st3v3nhunt** [comment]
New issue for using just one style of assertion #64

File: test/unit/server.js (hunk 1 above)
Comment (neilbmclaughlin) [inline test/unit/server.js:23]: Agreed but in my defence I was just playing with chai.expect.

**Pull request merged**